feedhenry/wendy-jenkins-s2i-configuration
# Jenkins slave poc

Added fix on top of @mikenairn 's poc, where the image itself would still be trying to load node-js scl, even though  only ruby scl was installed.

Added additional BuildConfig to template that takes the configuration in feedhenry/wendy-jenkins-s2i-configuration:jenkins_slave_poc slave-ruby folder, builds it and adds it to correctly tagged image stream.

This way, after updating wendy-jenkins-configuration, we just need to trigger image-rebuild inside openshift project, no need to manage docker registries or mannually push.

To verify:

1. Create new project with the jenkins template.
2. Wait for both jenkins and jenkins-slave-ruby-builds to finish
3. login to jenkins and create new pipeline-job
```
node ("ruby") {
   sh 'ruby --version'
}
```
4. run the job
5. you should see something like this in console:
```
Started by user Jenkins Admin
[Pipeline] node
Resuming build at Wed Jan 11 09:21:51 EST 2017 after Jenkins restart
Queue item for node block in testruby #1 is missing (perhaps JENKINS-34281); rescheduling
Running on jenkins-slave-ruby-d8fd2f35cfbb3 in /tmp/workspace/testruby
[Pipeline] {
[Pipeline] sh
[testruby] Running shell script
+ ruby --version
ruby 2.3.1p112 (2016-04-26 revision 54768) [x86_64-linux]
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
Finished: SUCCESS
```
